Data/DList.hs

```diff
@@ -225,6 +225,14 @@ map          :: (a -> b) -> DList a -> DList b
 map f        = foldr (cons . f) empty
 {-# INLINE map #-}
 
+-- | /O(n)/. Intersperses an element over difference lists.
+intersperse :: a -> DList a -> DList a
+intersperse sep = fromList . List.intersperse sep . toList
```

Comment:
Hmm. Sorry, but I don't think this is a good addition to the library.

I try to keep only functions that don't “break” the representation, e.g. by forcing the `DList` to a list and recreating a new `DList`. Notice that there are no other functions with `fromList . f . toList` for some `f`.

First, it's easy to see the type `a -> DList a -> DList a` and assume that the thunks of the `DList` argument are preserved, when, in fact, they are not. In other words, this function could be considered misleading. I'd rather somebody explicitly use at least one of the conversions, so that it's clear what's happening. (There are a number of functions defined using either `toList` or `fromList` for convenience, but I don't think it's a good idea to have a function that does both.)

Second, when one uses a `DList`, typically, I think one would use `fromList` to introduce it, then apply a number of functions, and then use `toList` (or similar) to eliminate the `DList`. I don't think one would, given a `DList`, convert it to a list just to apply `intersperse`, and then convert it back to a `DList`. That would probably defeat the purpose of using a `DList` in the first place. (If I'm wrong, I'd love to see an explanation of why. 😄)